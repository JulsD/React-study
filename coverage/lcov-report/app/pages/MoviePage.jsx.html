<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for app/pages/MoviePage.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> / <a href="index.html">app/pages</a> MoviePage.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>5/10</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">21.43% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>3/14</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>4/8</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>5/10</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
import { connect } from 'react-redux';
import { findMovie } from '../action';
&nbsp;
import {
  Header,
  Footer,
  SearchResult,
  HeaderNav,
  MovieBox,
  HeaderFooter,
  MovieInfo,
  NavMenu,
  MoviesList,
  Loader
} from './../components';
&nbsp;
export class MoviePage extends React.Component {
&nbsp;
  constructor(props) {
    super(props);
    this.state = {}
  }
&nbsp;
  searchMovieById(movieId){
    this.props.findMovie(movieId);
  }
&nbsp;
  componentDidMount (){
    this.searchMovieById(this.props.match.params.id);
  }
&nbsp;
<span class="fstat-no" title="function not covered" >  co</span>mponentWillReceiveProps(nextProps) {
<span class="cstat-no" title="statement not covered" >    if(nextProps.match.params.id !== this.props.match.params.id) {</span>
<span class="cstat-no" title="statement not covered" >      this.searchMovieById(nextProps.match.params.id);</span>
    }
  }
&nbsp;
  render() {
    return (
      &lt;div&gt;
        &lt;Header&gt;
          &lt;HeaderNav&gt;
            &lt;NavMenu menu={['Search']}/&gt;
          &lt;/HeaderNav&gt;
          { this.props.movieInfo.movie ?  &lt;MovieBox/&gt;  : (<span class="branch-1 cbranch-no" title="branch not covered" >this.props.isLoading ? &lt;Loader /&gt; : '')</span>}
        &lt;/Header&gt;
        &lt;HeaderFooter&gt;
          { this.props.movieInfo.movie ?  &lt;MovieInfo/&gt;  : (<span class="branch-1 cbranch-no" title="branch not covered" >this.props.isLoading ? &lt;Loader /&gt; : '')</span> }
        &lt;/HeaderFooter&gt;
        &lt;SearchResult&gt;
          {this.props.movieInfo.movie ? &lt;MoviesList movies={this.props.movieInfo.moviesByDir}/&gt; : (<span class="branch-1 cbranch-no" title="branch not covered" >this.props.isLoading ? &lt;Loader /&gt; : '')</span>}
        &lt;/SearchResult&gt;
        &lt;Footer /&gt;
      &lt;/div&gt;
    );
  }
&nbsp;
}
&nbsp;
function <span class="fstat-no" title="function not covered" >actions(</span>dispatch){
<span class="cstat-no" title="statement not covered" >  return {</span>
    findMovie: <span class="fstat-no" title="function not covered" >(.</span>..params) =&gt; <span class="cstat-no" title="statement not covered" >dispatch(findMovie(params))</span>
  }
}
&nbsp;
function <span class="fstat-no" title="function not covered" >mapProps(</span>state) {
<span class="cstat-no" title="statement not covered" >  return {</span>
    movieInfo: state.movie,
    isLoading: state.isLoading.movie
  }
}
&nbsp;
export default connect(mapProps, actions)(MoviePage);</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Thu Nov 09 2017 00:12:39 GMT+0100 (CET)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
