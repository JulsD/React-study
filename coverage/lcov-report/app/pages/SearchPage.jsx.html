<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for app/pages/SearchPage.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> / <a href="index.html">app/pages</a> SearchPage.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">47.83% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>11/23</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">38.89% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>7/18</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">44.44% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>4/9</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">47.83% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>11/23</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81
82
83
84
85
86
87
88
89
90
91
92
93
94
95
96
97
98
99
100
101
102
103</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
import sortBy from 'lodash/sortBy';
import reverse from 'lodash/reverse';
import * as queryString from 'query-string';
import { connect } from 'react-redux';
import { searchByTitle, searchByDirector } from '../action';
&nbsp;
import {
  Header,
  Footer,
  SearchResult,
  HeaderNav,
  SearchBox,
  HeaderFooter,
  SearchNav,
  SearchSort,
  SearchSum,
  MoviesList,
  EmptySearch,
  Loader
} from './../components';
&nbsp;
export class SearchPage extends React.Component {
&nbsp;
  constructor(props) {
    super(props);
    this.state = {}
  }
&nbsp;
  searchMoviesByParam(query){
    const searchQuery = queryString.parse(query);
      <span class="missing-if-branch" title="if path not taken" >I</span>if(searchQuery['title']) {
<span class="cstat-no" title="statement not covered" >        this.props.searchByTitle(searchQuery['title']);</span>
     } else <span class="missing-if-branch" title="if path not taken" >I</span>if (searchQuery['director']) {
<span class="cstat-no" title="statement not covered" >       this.props.searchByDirector(searchQuery['director']);</span>
     } else {
       return;
     }
  }
&nbsp;
  componentDidMount (){
    this.searchMoviesByParam(this.props.location.search);
  }
&nbsp;
<span class="fstat-no" title="function not covered" >  co</span>mponentWillReceiveProps(nextProps) {
<span class="cstat-no" title="statement not covered" >    if(nextProps.location.search !== this.props.location.search) {</span>
<span class="cstat-no" title="statement not covered" >      this.searchMoviesByParam(nextProps.location.search);</span>
    }
  }
&nbsp;
  render() {
&nbsp;
    let searchResultBody = null;
    <span class="missing-if-branch" title="if path not taken" >I</span>if(this.props.search.movies &amp;&amp; <span class="branch-1 cbranch-no" title="branch not covered" >!this.props.isLoading)</span>{
<span class="cstat-no" title="statement not covered" >      if(this.props.search.movies.length &gt; 0) {</span>
<span class="cstat-no" title="statement not covered" >        searchResultBody = &lt;MoviesList movies={reverse(sortBy(this.props.search.movies, [this.props.sortBy]))}/&gt;;</span>
      } else {
<span class="cstat-no" title="statement not covered" >        searchResultBody = &lt;EmptySearch /&gt;;</span>
      }
    } else <span class="missing-if-branch" title="if path not taken" >I</span>if (this.props.isLoading) {
<span class="cstat-no" title="statement not covered" >      searchResultBody = &lt;Loader /&gt;</span>
    }
&nbsp;
    return (
      &lt;div&gt;
        &lt;Header&gt;
          &lt;HeaderNav /&gt;
          &lt;SearchBox /&gt;
        &lt;/Header&gt;
        &lt;HeaderFooter&gt;
          &lt;SearchNav&gt;
            {this.props.search.movies ? <span class="branch-0 cbranch-no" title="branch not covered" >&lt;SearchSum/&gt; </span>: ''}
            {this.props.search.movies ? <span class="branch-0 cbranch-no" title="branch not covered" >&lt;SearchSort /&gt; </span>: ''}
          &lt;/SearchNav&gt;
        &lt;/HeaderFooter&gt;
        &lt;SearchResult&gt;
          {searchResultBody}
        &lt;/SearchResult&gt;
        &lt;Footer /&gt;
      &lt;/div&gt;
    );
  }
&nbsp;
}
&nbsp;
&nbsp;
function <span class="fstat-no" title="function not covered" >actions(</span>dispatch){
<span class="cstat-no" title="statement not covered" >  return {</span>
    searchByTitle: <span class="fstat-no" title="function not covered" >(.</span>..params) =&gt; <span class="cstat-no" title="statement not covered" >dispatch(searchByTitle(params)),</span>
    searchByDirector: <span class="fstat-no" title="function not covered" >(.</span>..params) =&gt; <span class="cstat-no" title="statement not covered" >dispatch(searchByDirector(params))</span>
  }
}
&nbsp;
function <span class="fstat-no" title="function not covered" >mapProps(</span>state) {
<span class="cstat-no" title="statement not covered" >  return {</span>
    sortBy: state.sort,
    search: state.search,
    isLoading: state.isLoading.search
  }
}
&nbsp;
export default connect(mapProps, actions)(SearchPage);
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Thu Nov 09 2017 00:12:39 GMT+0100 (CET)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
