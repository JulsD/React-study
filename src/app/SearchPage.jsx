import React from 'react';

import Header from './components/Header/Header';
import SearchResult from './components/SearchResult/SearchResult';
import Footer from './components/Footer/Footer';

import common from './../styles/common.css';

class SearchPage extends React.Component {

  constructor(props) {
    super(props);

    this.state = {movieOpend: false};
  }

  render() {
    const movieOpend = this.state.movieOpend;
    const movies = [1,2,3,4,5];
    return (
      <div className = {common.typo}>
        <Header movieOpend={movieOpend} emptyResults = {movies ? false : true}/>
        <SearchResult movies = {movies}/>
        <Footer />
      </div>
    );
  }

}

export default SearchPage;
