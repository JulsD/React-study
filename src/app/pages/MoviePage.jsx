import React from 'react';
import { connect } from 'react-redux';
import { findMovie } from '../action';

import {
  Header,
  Footer,
  SearchResult,
  HeaderNav,
  MovieBox,
  HeaderFooter,
  MovieInfo,
  NavMenu,
  MoviesList,
  Loader
} from './../components';

export class MoviePage extends React.Component {

  constructor(props) {
    super(props);
    this.state = {}
  }

  searchMovieById(movieId){
    this.props.findMovie(movieId);
  }

  componentDidMount (){
    this.searchMovieById(this.props.match.params.id);
  }

  componentWillReceiveProps(nextProps) {
    if(nextProps.match.params.id !== this.props.match.params.id) {
      this.searchMovieById(nextProps.match.params.id);
    }
  }

  render() {
    return (
      <div>
        <Header>
          <HeaderNav>
            <NavMenu menu={['Search']}/>
          </HeaderNav>
          { this.props.movieInfo.movie ?  <MovieBox/>  : (this.props.isLoading ? <Loader /> : '')}
        </Header>
        <HeaderFooter>
          { this.props.movieInfo.movie ?  <MovieInfo/>  : (this.props.isLoading ? <Loader /> : '') }
        </HeaderFooter>
        <SearchResult>
          {this.props.movieInfo.movie ? <MoviesList movies={this.props.movieInfo.moviesByDir}/> : (this.props.isLoading ? <Loader /> : '')}
        </SearchResult>
        <Footer />
      </div>
    );
  }

}

function actions(dispatch){
  return {
    findMovie: (...params) => dispatch(findMovie(params))
  }
}

function mapProps(state) {
  return {
    movieInfo: state.movie,
    isLoading: state.isLoading.movie
  }
}

export default connect(mapProps, actions)(MoviePage);