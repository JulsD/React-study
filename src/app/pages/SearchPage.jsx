import React from 'react';

import { 
  Header, 
  Footer, 
  SearchResult, 
  HeaderNav, 
  SearchBox, 
  HeaderFooter, 
  SearchNav
} from './../components';
import common from './../../styles/common.css';

class SearchPage extends React.Component {

  constructor(props) {
    super(props);
  }

  render() {
    const movies = [];
    let searchResultBody = null;
    if (movies) {
      searchResultBody = <MoviesList movies={movies}/>;
    } else {
     searchResultBody = <EmptySearch />;
    }
    return (
      <div className = {common.typo}>
        <Header>
          <HeaderNav />
          <SearchBox />
        </Header>
        <HeaderFooter>
          <SearchNav />
        </HeaderFooter>
        <SearchResult>
          {searchResultBody}
        </SearchResult>
        <Footer />
      </div>
    );
  }

}

export default SearchPage;
