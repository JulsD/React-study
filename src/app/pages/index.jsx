export {default as MoviePage} from './MoviePage';
export {default as SearchPage} from './SearchPage';