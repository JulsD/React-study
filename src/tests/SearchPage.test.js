import React from 'react';
import { SearchPage } from '../app/pages/SearchPage';

test('Logo should render a <div>', () => {
  const testStore = {
    search: [1,2,3],
    sortBy: 'test',
    isLoading: false,
    location: {search: 'test'},
    searchByTitle: () => true,
    searchByDirector: () => true
  }
  const wrapper = shallow(
    <SearchPage search={testStore.search}
                sortBy={testStore.sortBy}
                isLoading={testStore.isLoading}
                location={testStore.location}
                searchByTitle={testStore.searchByTitle}
                searchByDirector={testStore.searchByDirector}/>
  );
  expect(wrapper).toMatchSnapshot();
});
