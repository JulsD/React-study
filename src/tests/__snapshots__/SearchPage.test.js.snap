// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo should render a <div> 1`] = `
<div>
  <Header>
    <Nav />
    <ReduxForm />
  </Header>
  <HeaderFooter>
    <SearchNav />
  </HeaderFooter>
  <SearchResult />
  <Footer />
</div>
`;
